Adapters/Vungle/ISVungleAdapter/ISVungleBannerAdapterRouter.m

```diff
@@ -0,0 +1,205 @@
+//
+//  ISVungleBannerAdapterRouter.m
+//  ISVungleAdapter
+//
+//  Copyright © 2020 ironSource. All rights reserved.
+//
+
+#import "ISVungleBannerAdapterRouter.h"
+#import "ISVungleAdapter.h"
+
+@interface ISVungleBannerAdapterRouter ()
+
+@property (nonatomic, strong, nullable) NSString *bidPayload;
+
+@end
+
+@implementation ISVungleBannerAdapterRouter
+
+- (instancetype)initWithPlacementID:(NSString *)placementID
+                      parentAdapter:(ISVungleAdapter *)parentAdapter
+                           delegate:(id<ISBannerAdapterDelegate>)delegate {
+    if (self = [super init]) {
+        _placementID = placementID;
+        _parentAdapter = parentAdapter;
+        _delegate = delegate;
+        _bannerState = UNKNOWN;
+        _bidPayload = nil;
+        _bannerSize = BannerSizeRegular;
+    }
+
+    return self;
+}
+
+- (void)loadBannerAd {
+    self.bannerAd = [[VungleBanner alloc] initWithPlacementId:self.placementID size:self.bannerSize];
+    self.bannerAd.delegate = self;
+    self.bannerAd.enableRefresh = NO;
+    self.adView = [[UIView alloc] initWithFrame:[self getAdViewRect:self.bannerSize]];
+
+    // Because there's no auto-caching in 7.0 VungleAds,the Ad would
+    // never be ready until a load call is made. We don't need to
+    // check [VungleBanner canPlayAd] here.
+    [self.bannerAd load:self.bidPayload];
+}
+
+- (void)showBannerAd {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.bannerAd presentOn:self.adView];
+    });
+    self.bannerState = SHOWING;
+}
+
+- (void)setBidPayload:(NSString * _Nullable)bidPayload {
+    self.bidPayload = bidPayload;
```

Comment:
@YueVungle I am getting a crash here. It seems like you set `bidPayload` to `self.bidPayload` here will calling method (`(void)setBidPayload`) circularly？

Reply:
@ashinagawa Thanks a lot for pointing out this issue, because the bidding ad cannot work until the dashboard was configured before, I don't find the issue.Updated and tested that no crash happened. :)